app/models/miq_queue.rb

```diff
@@ -246,6 +246,23 @@ def self.submit_job(options)
     put(options)
   end
 
+  # For services with a dedicated queue_name and worker, such as reporting, high priority work is
+  # run in the 'generic' queue with HIGH_PRIORITY so priority workers or even generic workers can
+  # pick it up immediately. Anything but high priority work or higher is run in the service's
+  # queue, such as 'reporting', and the dedicated worker will process this message following
+  # priority processing.
+  #
+  # Why? For example, with 'reporting', reports are often started by a user who is actively waiting
+  # for the result so we need a way to expedite their request in workers who deal with high
+  # priority work.  At the same time, if a user isn't actively waiting for a report, it should be
+  # handled by reporting workers and not sit behind other items in the generic queue as it's far
+  # easier to scale these workers up and down as needed.
+  #
+  # TODO: Review if other services in submit_job, such as event/smart proxy should follow this pattern.
+  def self.queue_name_for_priority_service(service, priority)
+    MiqQueue.lower_priority?(priority || NORMAL_PRIORITY, HIGH_PRIORITY) ? service.to_s : "generic"
```

Comment:
This might be easier to read, but it's really up to you:

```suggestion
    (priority.nil? || MiqQueue.lower_priority?(priority, HIGH_PRIORITY)) ? service.to_s : "generic"
```

Reply:
done